### PR TITLE
[12.0] [ADD] sale_order_global_stock_route: New module

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT22057

### CODE REVIEW

File: sale_order_global_stock_route/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Sale Order Global Stock Route',
+    'summary': 'Add the possibility to choose '
+               'one warehouse path for an order',
+    'version': '12.0.1.0.1',
```
Comment (sergio-teruel) [inline sale_order_global_stock_route/__manifest__.py]: Change version to 12.0.1.0.0

File: sale_order_global_stock_route/models/stock_order.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    route_id = fields.Many2one(
+        comodel_name='stock.location.route',
+        string='Route',
+        domain=[('sale_selectable', '=', True)],
+        ondelete='restrict',
```
Comment (sergio-teruel) [inline sale_order_global_stock_route/models/stock_order.py]: This field only is a helper... do not apply restrict

File: sale_order_global_stock_route/views/sale_order_views.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <field name="payment_term_id" position="after">
+                <field name="route_id"/>
+            </field>
+            <field name="order_line" position="attributes">
+                <attribute name="context">{'default_route_id': route_id}</attribute>
```
Comment (sergio-teruel) [inline sale_order_global_stock_route/views/sale_order_views.xml]: Please user fields_view_get to add a context to void to break inheritence chain..
You can see this link https://github.com/odoo/odoo/blob/f228ae0ce5e635a2c4bcd8d1dd85b14e67e85640/odoo/models.py#L1446

File: sale_order_global_stock_route/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+from . import stock_order
```
Comment (sergio-teruel) [inline sale_order_global_stock_route/models/__init__.py]: I thinlk you must rename to sale_order

File: sale_order_global_stock_route/views/sale_order_views.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2020 Tecnativa - Carlos Roca -->
+
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <field name="payment_term_id" position="after">
+                <field name="route_id"/>
```
Comment (sergio-teruel) [inline sale_order_global_stock_route/views/sale_order_views.xml]: Please add security group "sale_stock.group_route_so_lines" as in sale_order_line

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
@sergio-teruel Changes done

File: sale_order_global_stock_route/i18n/es.po [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
+# * sale_order_global_stock_route
+#
+# Translators:
+# Carlos Roca <carlos.roca@tecnativa.com>, 2020
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 13.0\n"
```
Comment (manuelcalerosolis) [inline sale_order_global_stock_route/i18n/es.po]: Must be 12.0

**Review by manuelcalerosolis — CHANGES_REQUESTED** [review]

File: sale_order_global_stock_route/readme/CONTRIBUTORS.rst [hunk 7]
```diff
@@ -0,0 +1 @@
+* Carlos Roca (Tecnativa) <carlos.roca@tecnativa.com>
```
Comment (manuelcalerosolis) [inline sale_order_global_stock_route/readme/CONTRIBUTORS.rst]: * `Tecnativa <https://www.tecnativa.com>`_:

  * Carlos Roca (Tecnativa) <carlos.roca@tecnativa.com>

**Review by manuelcalerosolis — CHANGES_REQUESTED** [review]

File: sale_order_global_stock_route/readme/CONFIGURE.rst [hunk 8]
```diff
@@ -0,0 +1,2 @@
+For use this module you have to activate the option Settings > Inventory
+> Warehouse > Multi-Step Routes
```
Comment (manuelcalerosolis) [inline sale_order_global_stock_route/readme/CONFIGURE.rst:2]: I think it should contain the following comment:

In the user profile you must select this option "Enable Route on Sales Order Line" in the "Technical Settings" section

**Review by manuelcalerosolis — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
@manuelcalerosolis Changes done

**Review by sergio-teruel — APPROVED** [review]

File: sale_order_global_stock_route/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Sale Order Global Stock Route',
+    'summary': 'Add the possibility to choose '
+               'one warehouse path for an order',
+    'version': '12.0.1.0.0',
+    'development_status': 'Alpha',
```
Comment (pedrobaeza) [inline sale_order_global_stock_route/__manifest__.py]: Why alpha? Shouldn't this be `beta` once at least 2 of you have reviewed it?

Comment (CarlosRoca13) [reply sale_order_global_stock_route/__manifest__.py]: Change done

**Review by pedrobaeza — APPROVED** [review]

**Comment by rousseldenis** [comment]
@pedrobaeza Travis is red...

**Comment by pedrobaeza** [comment]
I have seen and it was a glitch on only one build. That was why I have launched the command.

**Pull request merged**